editor/store/selectors.js

```diff
@@ -303,7 +303,22 @@ export function isEditedPostSaveable( state ) {
 	return (
 		!! getEditedPostTitle( state ) ||
 		!! getEditedPostExcerpt( state ) ||
-		!! getEditedPostContent( state )
+		! isEditedPostEmpty( state )
+	);
+}
+
+/**
+ * Returns true if the edited post has content. A post has content if it has at
+ * least one block or otherwise has a non-empty content property assigned.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {boolean} Whether post has content.
+ */
+export function isEditedPostEmpty( state ) {
+	return (
+		! getBlockCount( state ) &&
+		! getEditedPostAttribute( state, 'content' )
```

Comment:
I wonder if we should leverage `isUnmodifiedDefaultBlock` to consider a post with only empty paragraphs as an empty post as well.

Reply:
>I wonder if we should leverage `isUnmodifiedDefaultBlock` to consider a post with only empty paragraphs as an empty post as well.

We could, but for all intents and purposes, such a post would not technically be empty, as emptiness in our case is that the server-side forbids post saves unless content is a non-empty string. Even if all blocks are empty paragraphs, the demarcations satisfy the requirement. It's also a bit more complex if we not only check the block count, but that each is a non-empty default type.